crates/router/src/core/webhooks.rs

```diff
@@ -1078,6 +1077,25 @@ pub async fn webhooks_core<W: types::OutgoingWebhookType, Ctx: PaymentMethodRetr
 
     logger::info!(process_webhook=?process_webhook_further);
 
+    let object_reference_id = connector
+        .get_webhook_object_reference_id(&request_details)
+        .switch()
+        .attach_printable("Could not find object reference id in incoming webhook body")?;
+
+    let merchant_connector_account = match merchant_connector_account {
+        Some(merchant_connector_account) => merchant_connector_account,
+        None => {
+            helper_utils::get_mca_from_object_reference_id(
+                &*state.clone().store,
+                object_reference_id,
+                &merchant_account,
+                &connector_name,
+                &key_store,
+            )
+            .await?
+        }
+    };
```

Comment:
What if event not supported doesn't have object reference id, Doesn't it still throw 4xx ? Shouldn't we ignore event not supported webhooks or Should we move this after `if process_webhook_further`